### PR TITLE
[KYUUBI #2605] Make SQLOperationListener configurable

### LINKED ISSUE #2605: [Improvement] Make SQLOperationListener configurable
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

https://github.com/apache/incubator-kyuubi/blob/7f8369bf18f8e37b31818e55877827c725696c89/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala#L83-L84

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by cxzl25**
### _Why are the changes needed?_
close #2605

`kyuubi.engine.spark.register.operation.listener`

> When set to true, Spark engine registers a SQLOperationListener before executing the statement, logs a few summary statistics when each stage completes.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by ulysses-you** [comment]
@cxzl25 please resolve the conflicts, thank you

**Review by ulysses-you — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala [hunk 1]
```diff
@@ -48,7 +48,14 @@ class ExecuteStatement(
   private val operationLog: OperationLog = OperationLog.createOperationLog(session, getHandle)
   override def getOperationLog: Option[OperationLog] = Option(operationLog)
 
-  private val operationListener: SQLOperationListener = new SQLOperationListener(this, spark)
+  private val conf: KyuubiConf = getSession.sessionManager.getConf
+
+  private val operationListener: Option[SQLOperationListener] =
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:53]: we can make it settable via sql command

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -1499,4 +1499,12 @@ object KyuubiConf {
       .version("1.6.0")
       .regexConf
       .createOptional
+
+  val ENGINE_SPARK_REGISTER_OPERATION_LISTENER: ConfigEntry[Boolean] =
+    buildConf("kyuubi.engine.spark.register.operation.listener")
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1504]: kyuubi.operation.spark.listener.enabled

**Comment by yaooqinn** [comment]
Can we also add an unit test for this?

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala [hunk 3]
```diff
@@ -42,13 +41,19 @@ class ExecuteStatement(
     override val statement: String,
     override val shouldRunAsync: Boolean,
     queryTimeout: Long,
-    incrementalCollect: Boolean)
+    incrementalCollect: Boolean,
+    operationSparkListenerEnabled: Boolean)
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:45]: refer to progressEnable？

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala (hunk 1 above)
Comment (ulysses-you) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:56]: @cxzl25 seems this comment has not been addressed ? can you take a look ?

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala (hunk 1 above)
Comment (cxzl25) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:56]: It should be solved, I add set true/false in UT to test.
Thanks for your review.

**Pull request closed without merging**